tools/integration_tests_runner.py

```diff
@@ -2,136 +2,130 @@
 
 
-class IntegrationRunner:
-    def __init__(self):
-        self.process_cli()
+def setup_logger(logger_name, log_file, level=logging.INFO):
+    l = logging.getLogger(logger_name)
+    formatter = logging.Formatter('%(msg)s')
+    file_handler = logging.FileHandler(log_file, mode='w')
+    file_handler.setFormatter(formatter)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
 
-        self.proc_count = multiprocessing.cpu_count()
-        logging.info("Number of processors is: {nproc}".format(nproc=self.proc_count))
+    l.setLevel(level)
+    l.addHandler(file_handler)
+    l.addHandler(stream_handler)
 
-        self.file_lock = Lock()
-        self.start_finish_lock = Lock()
-        self.tests = Queue()
-        self.key_postfix = ""
-        if self.user_resource_path:
-            self.key_postfix += ' --user_resource_path="{0}"'.format(self.user_resource_path)
-        if self.data_path:
-            self.key_postfix += ' --data_path="{0}"'.format(self.data_path)
 
+def execute_file(test, keys):
+    spell = ["{test} {keys}".format(test=test, keys=keys)]
+    logging.debug(">> {spell}".format(spell=spell))
 
-    def run_tests(self):
-        intermediate_tests = []
-        for exec_file in self.runlist:
-            intermediate_tests.extend(map(lambda x: (exec_file, x), self.get_tests_from_exec_file(exec_file, "--list_tests")[0]))
+    process = subprocess.Popen(spell,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               shell=True
+                               )
 
-        shuffle(intermediate_tests)
-        for test in intermediate_tests:
-            self.tests.put(test)
+    out, err = process.communicate()
+    result = process.returncode
 
-        with open(self.output, "w") as self.file, open("start-finish.log", "w") as self.start_finish_log:
-            self.run_parallel_tests()
+    return filter(None, out.split("\n")), err, result
 
 
-    def run_parallel_tests(self):
-        threads = []
+def exec_test(a_tuple):
+    test_file, test, key_postfix = a_tuple
+    keys = '"--filter={test}"'.format(test=test)
 
-        for i in range(0, self.proc_count):
-            thread = Thread(target=self.exec_tests_in_queue)
-            thread.start()
-            threads.append(thread)
+    if test_file in TEMPFOLDER_TESTS:
+        tmpdir = tempfile.mkdtemp()
+        keys = '{old_key} "--data_path={tmpdir}"'.format(old_key=keys, tmpdir=tmpdir)
+        logging.debug("Temp dir: {tmpdir}".format(tmpdir=tmpdir))
+    else:
+        keys = "{old_key}{resource_path}".format(old_key=keys, resource_path=key_postfix)
+        logging.debug("Setting user_resource_path and data_path to {resource_path}".format(resource_path=key_postfix))
 
-        for thread in threads:
-            thread.join()
+    log_start_finish(test_file, keys, start=True)
```

Comment:
Explicit != читабельно. Особенно в сценарии, где не выполняется никакой
значимой функциональности.
И почему две?

2016-06-16 11:43 GMT+03:00 therearesomewhocallmetim <
notifications@github.com>:

> In tools/integration_tests_runner.py
> <https://github.com/mapsme/omim/pull/3575#discussion_r67307652>:
>
> >
> > -        for thread in threads:
> > -            thread.join()
> > +    log_start_finish(test_file, keys, start=True)
>
> Можно, но тогда у нас добавится еще две функции-обертки. Мне показалось,
> что эксплицитно указать start или finish будет достаточно читабельно.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/3575/files/403f4e992cc2f19cc5406f315fcf267efb56ede4#r67307652>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACGjAcidiQd_uYZOi9YDbcASQWtnOnmsks5qMQy5gaJpZM4I2caF>
> .
>